### DISCUSSION TITLE
Use only the CNMF algorithm from CaImAn

### DISCUSSION THREAD

**Discussion by LeoJuh** [flatironinstitute/CaImAn#1331] [General]
My lab is using a python package called [to](https://gitlab.iit.it/fellin-public/cite-on) for cell identification. The identified cells is thereafter   feeded into the CaImAn software.

A problem with this is that since cite_on and CaImAn use two different versions of tensorflow, they can´t be used in the same conda environment, which makes things a bit tricker. 

I therefore would like to ask if anyone knows a way to make a CaImAn "lite" version which only contains the code to execute the CNMF function?

**Comment by kushalkolar** [comment]
Use separate environments for each?

**Reply by LeoJuh** [reply]
Yes, but is there a way to run them in the same environment? To trim down CaImAn to only its CNMF functionality?

**Reply by kushalkolar** [reply]
You can try to isolate the CNMF code, but that is probably harder than using 2 environments for your pipeline.

Are you doing online analysis that requires immediately sending data from `cite_on` to CNMF?

**Reply by LeoJuh** [reply]
No, so it is not the biggest issue. However, it would be nice to implement since we are going to use our software with caiman's CNMF function in a lot of stuff in the future.

Do you have any advice for how to isolate the CNMF code?

**Reply by kushalkolar** [reply]
You can possibly start with removing the tensorflow imports and then begin triaging from there until you end up with a codebase where cnmf still works.

Note that tensorflow is used for component evaluation with the CNN, so if you need that then you need tensorflow.

**Reply by LeoJuh** [reply]
Thanks for the advice!

**Comment by pgunn** [comment]
Note that if all you need is the Oasis part of CNMF, Johannes has a separate repo that just has those parts of the codebase:

https://github.com/j-friedrich/OASIS

I've never used that codebase, but it looks like it should probably work.

Note as well that tensorflow is something we're hoping to replace with pytorch at some point (which might or might not make the underlying issue easier.

**Comment by pgunn** [comment]
@LeoJuh One additional note - I looked at the repo and you're pinning tensorflow 1.13 - in the longer-term this is going to cause problems as supportable cuda versions will limit your ability to work with current GPU hardware. At some point you'll probably want to port cite-on to use tensorflow 2.x (or move to torch, or one of the other options) just to continue to be able to use it.

**Reply by LeoJuh** [reply]
Ok makes sense, thanks for the heads up!
